### PR TITLE
TOOK-47 Import Assets Directly

### PR DESCRIPTION
**Pull request by fcturan20**
[TOOK-47](https://oyunteknolojileri.atlassian.net/browse/TOOK-47?atlOrigin=eyJpIjoiOWQyNzZjOTg5ZDI2NGEwZGEwNDNiOGM0NzQyN2M2YzQiLCJwIjoiaiJ9)

### CODE REVIEW

File: Editor/App.cpp [hunk 1]
```diff
@@ -682,18 +682,16 @@ namespace ToolKit
       // create a build dir if not exist.
       std::filesystem::create_directories(buildDir);
 
-      // Update project files in case of change.
-      #ifdef TK_DEBUG
+// Update project files in case of change.
+#ifdef TK_DEBUG
       static constexpr char buildConfig[] = "Debug";
```
Comment (afraidofdark) [inline Editor/App.cpp:687]: change both please "static const StringView buildConfig"
We are moving to StringView for const char*

File: Editor/App.cpp [hunk 2]
```diff
@@ -980,7 +978,10 @@ namespace ToolKit
 
           // Execute command
           result = ExecSysCommand(cmd.c_str(), false, false);
-          assert(result != -1);
+          if (result != 0)
+          {
+            GetLogger()->WriteConsole(LogType::Warning, "Import failed!");
```
Comment (afraidofdark) [inline Editor/App.cpp:983]: Error is better suited

File: Editor/UI.cpp [hunk 3]
```diff
@@ -809,15 +809,57 @@ namespace ToolKit
...
+          String ext;
+          DecomposePath(file, nullptr, nullptr, &ext);
+
+          if (ext == ".png" || ext == ".hdri")
+          {
+            if (ImportData.activeView->GetPath().find(TexturePath("")) !=
+                String::npos)
+            {
+              // Both HDRIs and PNGs stored in Textures/ root folder
+              std::filesystem::copy(
```
Comment (afraidofdark) [inline Editor/UI.cpp:846]: does this means any texture will be stored in the root ?
I rather have them stored in the sub directory if user drop them in sub directories. This was the requirement of the ticket. However, surly sub directory must be texture.

File: Editor/UI.h [hunk 4]
```diff
@@ -20,6 +20,8 @@ namespace ToolKit
 
   namespace Editor
   {
+    class FolderView;
```
Comment (afraidofdark) [inline Editor/UI.h:23]: remove this line

File: Editor/UI.h [hunk 5]
```diff
@@ -148,7 +150,8 @@ namespace ToolKit
         bool overwrite        = false;
         StringArray files;
         String subDir;
-        float scale = 1.0f;
+        float scale            = 1.0f;
+        FolderView* activeView = nullptr;
```
Comment (afraidofdark) [inline Editor/UI.h:154]: class FolderView* activeView
Also, struct members should start with capital letters
ShowImportWindow
Overwrite
Files ...

File: Editor/main.cpp [hunk 6]
```diff
@@ -280,8 +280,17 @@ namespace ToolKit
 
       if (e.type == SDL_DROPFILE)
       {
-        UI::ImportData.files.push_back(e.drop.file);
-        UI::ImportData.showImportWindow = true;
+        const FolderWindowRawPtrArray& assetBrowsers =
```
Comment (afraidofdark) [inline Editor/main.cpp:283]: Move logic out of main.cpp !
Only signla & data passing. All the logic must be performed within Editor code.

**Review by afraidofdark — CHANGES_REQUESTED** [review]

File: Editor/UI.cpp (hunk 3 above)
Comment (fcturan20) [reply Editor/UI.cpp:846]: No, textures will be stored in the dropped subdirectory. I added how it works in the ticket.

**Review by afraidofdark — APPROVED** [review]

**Pull request merged**